### DISCUSSION TITLE
send_keys_slowly()

### DISCUSSION THREAD

**Discussion by pr3z3ntbhaskar** [seleniumbase/SeleniumBase#1249] [Ideas]
Hi,

First of all Thanks for great package. It takes selenium adoption to next level. :)

To test autocomplete features ( entry fields, searching in dropdown list), we have to send input char-by-char.
It would be nice to implement send_keys_slowly() as a wrapper to send_keys() with time_interval param.
Normally, Iam doing as below now.

def send_keys_slowly(self, element, text, time_interval = '0.02'):
        for onechar in text:
            element.send_keys( onechar)
            self.sleep( time_interval)

self.send_keys_slowly( self.find_element( "#input_field"), 'search param'))

P.S. 'element' can be locator etc as in send_keys().

Thanks,
Bhaskar

**Comment by mdmintz** [comment]
I think it would be easier to do the following then adding a whole new method for it:

```python
from seleniumbase import BaseCase

class DemoSiteTests(BaseCase):
    def test_demo_site(self):
        self.open("https://seleniumbase.io/demo_page")
        for key in "This is Automated":
            self.add_text("#myTextInput", key)
            self.sleep(0.1)
```

**Reply by pr3z3ntbhaskar** [reply]
Makes sense. Thanks a lot. :)
